### PR TITLE
Add external reference to updateWarehouse

### LINKED ISSUE #13342: Add external reference to updateWarehouse
Currently, the `updateWarehouse` mutations takes only two parameters: id and input.
What I'd like to pass is the `externalReference` to avoid unnecessary query for ID and then do the update.

### PR DESCRIPTION
**Pull request by Smit-Parmar**
I want to merge this change because
closes #13342 

# Impact

* [ ] New migrations
* [ ] New/Updated API fields or mutations
* [ ] Deprecated API fields or mutations
* [ ] Removed API types, fields, or mutations
* [ ] Documentation needs to be updated

# Pull Request Checklist

* [ ] Privileged queries and mutations are either absent or guarded by proper permission checks
* [ ] Database queries are optimized and the number of queries is constant
* [ ] Database migrations are either absent or optimized for zero downtime
* [ ] The changes are covered by test cases

### CODE REVIEW

**Review by fowczarek — APPROVED** [review]

File: CHANGELOG.md [hunk 1]
```diff
@@ -93,7 +93,8 @@ Shipping methods can be removed by the user after it has been assigned to a chec
   - Add `CheckoutLine.problems` field
   - Add `CheckoutSettingsInput` to `ChannelCreateInput` & `ChannelUpdateInput`
   - Add `checkoutSettings` field to `Channel`
-
+- Add `externalReference` to `updateWarehouse`. It will allow update warehouse by
+  external reference. - #13342 by @Smit-Parmar
```
Comment (zedzior) [inline CHANGELOG.md:97]: 1. Please rebase with latest `main` branch and move your changes to 3.16 section, as 3.15 has been already released.
2. You don't have to break the lines in `CHANGELOG` file.

File: saleor/graphql/warehouse/mutations/warehouse_update.py [hunk 2]
```diff
@@ -21,10 +22,14 @@ class Meta:
...
 
     class Arguments:
-        id = graphene.ID(description="ID of a warehouse to update.", required=True)
+        id = graphene.ID(description="ID of a warehouse to update.", required=False)
         input = WarehouseUpdateInput(
             required=True, description="Fields required to update warehouse."
         )
+        external_reference = graphene.String(
+            required=False,
+            description="External reference of an warehouse." + ADDED_IN_315,
```
[the remark this answers is not in the document]
Comment (zedzior) [reply saleor/graphql/warehouse/mutations/warehouse_update.py:31]: LGTM, just left some cosmetic remarks

File: CHANGELOG.md (hunk 1 above)
Comment (Smit-Parmar) [reply CHANGELOG.md:97]: @zedzior Please correct me If I'm wrong but not seeing any releases of 3.15 in repo.

Comment (zedzior) [reply CHANGELOG.md:97]: There is an alpha version already being tested, so all the new changes will go to 3.16. https://github.com/saleor/saleor/pull/13696

**Review by zedzior — APPROVED** [review]

**Review by maarcingebala — APPROVED** [review]

**Review by szczecha — APPROVED** [review]

**Pull request merged**